engines/ags/shared/gfx/allegro_bitmap.cpp

```diff
@@ -128,8 +128,24 @@ bool Bitmap::LoadFromFile(const char *filename) {
 
 	BITMAP *al_bmp = load_bitmap(filename, nullptr);
 	if (al_bmp) {
-		_alBitmap = al_bmp;
-		_isDataOwner = true;
+		if (al_bmp->format.bytesPerPixel == 3) {  // if bpp=24, convert to 32
+			Graphics::PixelFormat tmpPixelFormat;
+#ifdef SCUMM_LITTLE_ENDIAN
+			tmpPixelFormat = Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24);
+#else
+			tmpPixelFormat = Graphics::PixelFormat(4, 8, 8, 8, 8, 8, 16, 24, 0);
+#endif
```

Comment:
I think we might want to use `Graphics::PixelFormat(4, 8, 8, 8, 8, 16, 8, 0, 24)` on both little endian and big endian, as in https://github.com/scummvm/scummvm/blob/master/engines/ags/lib/allegro/surface.cpp#L438-L453

But that also make me think we might want to also convert to that format in the case where the generated BMP uses 4 bytes per pixel but a different pixel format. An similarly 2 bytes per pixel other than RGB565 should probably be converted to RGB565.

Reply:
Ok, I changed the function as requested